### PR TITLE
plugin aks has tags

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/kubernetesservice/aksHasTags.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+//var compareVersions = require('compare-versions');
+
+module.exports = {
+    title: 'Azure Kubernetes Servuce Has Tags',
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.js:6]:
```suggestion
    title: 'Azure Kubernetes Service Has Tags',
```

File: plugins/azure/kubernetesservice/aksHasTags.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+var helpers = require('../../../helpers/azure/');
+//var compareVersions = require('compare-versions');
+
+module.exports = {
+    title: 'Azure Kubernetes Service Has Tags',
+    category: 'Kubernetes Service',
+    domain: 'Containers',
+    description: 'Ensures the Azure Kubernetes Service Has Tags.',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    recommended_action: 'Modify Kubernetes service and add tags.',
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.js:11]:
```suggestion
    recommended_action: 'Modify AKS clusters and add tags.',
```

File: plugins/azure/kubernetesservice/aksHasTags.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            if (!managedClusters.data.length) {
+                helpers.addResult(results, 0, 'No existing Kubernetes clusters found', location);
+                return rcb();
+            }
+
+            for (let cluster of managedClusters.data) {
+                if (!cluster.id) continue;
+
+                if (cluster.tags && Object.entries(cluster.tags).length > 0){
+                    helpers.addResult(results, 0, 'Azure Kubernetes Service has tags associated', location, cluster.id);
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.js:41]:
```suggestion
                    helpers.addResult(results, 0, 'AKS cluster has tags', location, cluster.id);
```

File: plugins/azure/kubernetesservice/aksHasTags.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Azure Kubernetes Service does not have tags associated', location, cluster.id);
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.js:43]:
```suggestion
                    helpers.addResult(results, 2, 'AKS cluster does not have tags', location, cluster.id);
```

File: plugins/azure/kubernetesservice/aksHasTags.spec.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            };
+            const cache = createCache(null,null);
+            aksHasTags.run(cache, {}, callback);
+        })
+
+        it('should give failing result if Kubernetes Service does not have tags', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Azure Kubernetes Service does not have tags associated');
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.spec.js:146]:
```suggestion
                expect(results[0].message).to.include('AKS cluster does not have tags');
```

File: plugins/azure/kubernetesservice/aksHasTags.spec.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+            const cache = createCache(null,[listCluster[1]]);
+            aksHasTags.run(cache, {}, callback);
+        });
+
+        it('should give passing result if Kubernetes Service have tags', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Azure Kubernetes Service has tags associated');
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.spec.js:159]:
```suggestion
                expect(results[0].message).to.include('AKS cluster has tags');
```

File: plugins/azure/kubernetesservice/aksHasTags.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+//var compareVersions = require('compare-versions');
+
+module.exports = {
+    title: 'Azure Kubernetes Service Has Tags',
```
Comment (alphadev4) [inline plugins/azure/kubernetesservice/aksHasTags.js:6]:
```suggestion
    title: 'AKS cluster Has Tags',
```

Comment (alphadev4) [reply plugins/azure/kubernetesservice/aksHasTags.js:6]: Tags are added on clusters in AKS and file name as well

File: exports.js [hunk 8]
```diff
@@ -834,6 +834,7 @@ module.exports = {
         'rbacEnabled'                   : require(__dirname + '/plugins/azure/kubernetesservice/rbacEnabled.js'),
         'aksLatestVersion'              : require(__dirname + '/plugins/azure/kubernetesservice/aksLatestVersion.js'),
         'aksAgentVersion'               : require(__dirname + '/plugins/azure/kubernetesservice/aksAgentVersion.js'),
+        'aksClusterHasTags'                    : require(__dirname + '/plugins/azure/kubernetesservice/aksClusterHasTags.js'),
```
Comment (AkhtarAmir) [inline exports.js:837]:
```suggestion
        'aksClusterHasTags'              : require(__dirname + '/plugins/azure/kubernetesservice/aksClusterHasTags.js'),
```

File: exports.js [hunk 9]
```diff
@@ -834,6 +834,7 @@ module.exports = {
         'rbacEnabled'                   : require(__dirname + '/plugins/azure/kubernetesservice/rbacEnabled.js'),
         'aksLatestVersion'              : require(__dirname + '/plugins/azure/kubernetesservice/aksLatestVersion.js'),
         'aksAgentVersion'               : require(__dirname + '/plugins/azure/kubernetesservice/aksAgentVersion.js'),
+        'aksClusterHasTags'              : require(__dirname + '/plugins/azure/kubernetesservice/aksClusterHasTags.js'),
```
Comment (AkhtarAmir) [inline exports.js:837]:
```suggestion
        'aksClusterHasTags'             : require(__dirname + '/plugins/azure/kubernetesservice/aksClusterHasTags.js'),
```

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**